contracts/lbp/LBPManager.sol

```diff
@@ -234,6 +240,47 @@ contract LBPManager {
         vault.joinPool(lbp.getPoolId(), address(this), address(this), request);
     }
 
+    /**
+     * @dev start LBP just before one block
+     * @notice it can be invoked by anyone only once
+     */
+    function startLbp() external {
+        uint256 startTime;
+        uint256 blockTime = 5 minutes;
+        bool isSwapEnabled = lbp.getSwapEnabled();
+        (startTime, , ) = lbp.getGradualWeightUpdateParams();
+        require(
+            !isSwapEnabled && state == TaskState.Waiting,
+            "LBPManager: swapping already active"
+        );
+        require(
+            block.timestamp > startTime - blockTime,
+            "LBPManager: Only once block before start time"
+        );
+        state = TaskState.Started;
+        lbp.setSwapEnabled(true);
+    }
+
+    /**
+     * @dev ends LBP just after one block
+     * @notice it can be invoked by anyone only once
+     */
+    function endLbp() external {
+        uint256 endTime;
+        bool isSwapEnabled = lbp.getSwapEnabled();
+        (, endTime, ) = lbp.getGradualWeightUpdateParams();
+        require(
+            isSwapEnabled && state == TaskState.Started,
+            "LBPManager: swapping already active"
+        );
+        require(
+            block.timestamp > endTime,
+            "LBPManager: Only once block before start time"
```

Comment:
should be >=

incorrect grammar and unclear. should be "LBPManager: haven't reached endTime"?

Reply:
I have implemented the changes you requested.